### PR TITLE
Fix some warnings in ex2_obc_software

### PR DESCRIPTION
**Pull request by rcunrau**
The one that bugged me the most was "elif" incorrectly spelled as "else if" in system.h

### CODE REVIEW

File: include/os_portable.h [hunk 1]
```diff
@@ -74,6 +74,9 @@
 #ifndef PORTABLE_H
 #define PORTABLE_H
 
+#include <stddef.h>
```
Comment (Taylorr82) [inline include/os_portable.h:77]: This is actually code provided by TI, and changing it here will cause halcogen to overwrite it next time someone runs it and these changes may get squashed

File: main/system.h [hunk 2]
```diff
@@ -56,7 +56,7 @@
     #define UHF_SCI CSP_SCI  //UART2
 #if defined(IS_3U)
     #define GPS_SCI sciREG1  //UART3
-#else if defined(IS_2U)
+#elif defined(IS_2U)
```
Comment (Taylorr82) [inline main/system.h:59]: I see your point about style. Does this change anything? Just curious

**Review by Taylorr82 — CHANGES_REQUESTED** [review]

File: main/system.h (hunk 2 above)
Comment (rcunrau) [reply main/system.h:59]: #else and #elif have different semantics. The prepocessor warning was "extra characters ignored". The extra characters that are ignored are "if defined(IS_2U)", i.e. the body of the else wasn't actually conditional on IS_2U being defined.

File: include/os_portable.h (hunk 1 above)
Comment (rcunrau) [reply include/os_portable.h]: I just removed the changes to os_portable.h

**Review by arrooney — APPROVED** [review]

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**